src/lightning/pytorch/utilities/argparse.py

```diff
@@ -15,101 +15,36 @@
-) -> Union["pl.LightningDataModule", "pl.Trainer"]:
-    """Create an instance from CLI arguments. Eventually use variables from OS environment which are defined as
-    ``"PL_<CLASS-NAME>_<CLASS_ARUMENT_NAME>"``.
-
-    Args:
-        cls: Lightning class
-        args: The parser or namespace to take arguments from. Only known arguments will be
-            parsed and passed to the :class:`Trainer`.
-        **kwargs: Additional keyword arguments that may override ones in the parser or namespace.
-            These must be valid Trainer arguments.
-
-    Examples:
-
-        >>> from lightning.pytorch import Trainer
-        >>> parser = ArgumentParser(add_help=False)
-        >>> parser = Trainer.add_argparse_args(parser)
-        >>> parser.add_argument('--my_custom_arg', default='something')  # doctest: +SKIP
-        >>> args = Trainer.parse_argparser(parser.parse_args(""))
-        >>> trainer = Trainer.from_argparse_args(args, logger=False)
-    """
-    if isinstance(args, ArgumentParser):
-        args = cls.parse_argparser(args)
-
-    params = vars(args)
-
-    # we only want to pass in valid Trainer args, the rest may be user specific
-    valid_kwargs = inspect.signature(cls.__init__).parameters
-    trainer_kwargs = {name: params[name] for name in valid_kwargs if name in params}
-    trainer_kwargs.update(**kwargs)
-
-    return cls(**trainer_kwargs)
-
-
-def parse_argparser(cls: _ARGPARSE_CLS, arg_parser: Union[ArgumentParser, Namespace]) -> Namespace:
-    """Parse CLI arguments, required for custom bool types."""
-    args = arg_parser.parse_args() if isinstance(arg_parser, ArgumentParser) else arg_parser
-
-    types_default = {arg: (arg_types, arg_default) for arg, arg_types, arg_default in get_init_arguments_and_types(cls)}
-
-    modified_args = {}
-    for k, v in vars(args).items():
-        if k in types_default and v is None:
-            # We need to figure out if the None is due to using nargs="?" or if it comes from the default value
-            arg_types, arg_default = types_default[k]
-            if bool in arg_types and isinstance(arg_default, bool):
-                # Value has been passed as a flag => It is currently None, so we need to set it to True
-                # We always set to True, regardless of the default value.
-                # Users must pass False directly, but when passing nothing True is assumed.
-                # i.e. the only way to disable something that defaults to True is to use the long form:
-                # "--a_default_true_arg False" becomes False, while "--a_default_false_arg" becomes None,
-                # which then becomes True here.
-
-                v = True
-
-        modified_args[k] = v
-    return Namespace(**modified_args)
-
-
-def parse_env_variables(cls: _ARGPARSE_CLS, template: str = "PL_%(cls_name)s_%(cls_argument)s") -> Namespace:
+def _parse_env_variables(cls: _ARGPARSE_CLS, template: str = "PL_%(cls_name)s_%(cls_argument)s") -> Namespace:
```

Comment:
We don't need `_ARGPARSE_CLS` anymore

```suggestion


def _parse_env_variables(cls: Type, template: str = "PL_%(cls_name)s_%(cls_argument)s") -> Namespace:
```